### PR TITLE
2303 modal path navigation

### LINKED ISSUE #2303: withoutSync navigation for modalPathParam
When opening a modal with showModalPathInUrl enabled, the mfe path is shown as modalPathParam. If an internal navigation is triggered, modalPathParam is not updated to reflect the new route, so when doing a page refresh, the user ends up at the first opened modal path.

Expected result:
- [ ] There should be a way to update modalPathParam on internal route change.
- [ ] It should be possible to configure it via angular-support-lib/auto-navigation.

### PR DESCRIPTION
**Pull request by wdoberschuetz**
**Description**

Changes proposed in this pull request:
- update of ```modalPathParam``` now available in ```core```, ```client``` and ```client-frameworks-support/client-support-angular/projects/client-support-angular```

**Related issue(s)**
Resolves #2303 

**How to test**
1. Checkout the branch. You shall build ```core```, ```client``` and ```client-frameworks-support/client-support-angular/projects/client-support-angular```
2. Go to ```core/examples/luigi-example-angular/node_modules/@luigi-project``` and paste for each of the folders what you've built
3. For ```client-frameworks-support``` go to ```client-frameworks-support\client-support-angular\dist\client-support-angular``` and copy the folders and files to the ```core\examples\luigi-example-angular\node_modules\@luigi-project\client-support-angular```
4. For ```core``` go to ```core/public``` and copy the ```luigi.js``` and ```luigi.svelte.map.js``` to ```node_modules/@luigi-project/core```
5. For ```client``` go to ```client/public``` and copy the whole folder to ```node_modules/@luigi-project/client```

### CODE REVIEW

File: client-frameworks-support/client-support-angular/projects/client-support-angular/src/lib/service/luigi-auto-routing.service.ts [hunk 1]
```diff
@@ -111,16 +113,16 @@ export class LuigiAutoRoutingService implements OnDestroy {
...
         }
         console.debug('Calling fromVirtualTreeRoot for url ==> ' + url);
-        linkManager()
-          .fromVirtualTreeRoot()
-          .withoutSync()
-          .navigate(url);
+        linkManager().fromVirtualTreeRoot().withoutSync().navigate(url);
+      }
+      if (current.data.updateModalDataPath && current.data.path) {
+        let lm = linkManager();
```
Comment (rafalgamon) [inline client-frameworks-support/client-support-angular/projects/client-support-angular/src/lib/service/luigi-auto-routing.service.ts:124]: Why not `const`?

File: core/examples/luigi-example-angular/src/app/sample1/sample1.component.ts [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
   templateUrl: './sample1.component.html'
 })
 export class Sample1Component {
   constructor() {}
+
+  openModal() {
+    linkManager().openAsModal('/home');
+  }
+
+  // goTo() {
```
Comment (rafalgamon) [inline core/examples/luigi-example-angular/src/app/sample1/sample1.component.ts:15]: Why you leave comment?

**Comment by rafalgamon** [comment]
@wdoberschuetz  please ask @JohannesDoberer or @hardl about this warning in `Inclusive Language`

Match Found
Please consider an alternative to `master`. 
Possibilities include: `primary`, `main`, `leader`, `active`, `writer`

File: client-frameworks-support/client-support-angular/README.md [hunk 3]
```diff
@@ -1,27 +1,3 @@
-# ClientSupportAngular
+# How to proceed
```
Comment (alexandra-simeonova) [inline client-frameworks-support/client-support-angular/README.md:1]:
```suggestion
# ClientSupportAngular
```

**Review by alexandra-simeonova — APPROVED** [review]

File: core/src/services/routing.js [hunk 4]
```diff
@@ -525,6 +525,33 @@ class RoutingClass {
...
+        const queryParamIndex = location.hash.indexOf(queryParamSeparator);
+        if (queryParamIndex !== -1) {
+          url.hash = url.hash.slice(0, queryParamIndex);
+        }
+        url.hash = `${url.hash}${queryParamSeparator}${RoutingHelpers.encodeParams(params)}`;
+      } else {
+        url.search = `?${RoutingHelpers.encodeParams(params)}`;
+      }
+      
+      window.location = url.href;
```
Comment (JohannesDoberer) [inline core/src/services/routing.js:551]: Just replace the url or do we need to take care of the history?

**Review by JohannesDoberer — CHANGES_REQUESTED** [review]
Before we merge this PR we need to revert all changes in core/example/luigi-example-angular. This project is only an example how angular and luigi works together. I would say we keep it until it is approved because we can test the client-support-angular lib in this project.

File: core/src/services/routing.js [hunk 5]
```diff
@@ -525,6 +525,33 @@ class RoutingClass {
     return wc_container;
   }
 
+  updateModalDataInUrl(modalPath, modalParams) {
```
[the remark this answers is not in the document]
Comment (wdoberschuetz) [reply core/src/services/routing.js]: Unit tests provided.

File: core/src/services/routing.js (hunk 4 above)
Comment (wdoberschuetz) [reply core/src/services/routing.js]: Changed to ```pushState``` or ```replaceState``` depending on the case.

**Review by JohannesDoberer — APPROVED** [review]

**Pull request merged**